lib/pure/net.nim

```diff
@@ -1740,13 +1757,40 @@ proc sendTo*(socket: Socket, address: string, port: Port,
              data: string) {.tags: [WriteIOEffect].} =
   ## This proc sends `data` to the specified `address`,
   ## which may be an IP address or a hostname, if a hostname is specified
-  ## this function will try each IP of that hostname.
+  ## this function will try each IP of that hostname. Generally
+  ## for use with connection-less (UDP) sockets.
   ##
   ## If an error occurs an OSError exception will be raised.
   ##
   ## This is the high-level version of the above `sendTo` function.
   socket.sendTo(address, port, cstring(data), data.len, socket.domain)
 
+proc sendTo*(socket: Socket,
+             address: IpAddress, port: Port,
+             data: var string, size: int,
+             af: Domain = AF_INET, flags = 0'i32): int {.
+              discardable, tags: [WriteIOEffect].} =
+  ## This proc sends `data` to the specified `IpAddress`. 
+  ## Generally for use with connection-less (UDP) sockets. Also 
+  ## returns the number of bytes written.
+  ##
+  ## If an error occurs an OSError exception will be raised.
+  ##
+  ## This is the high-level version of the above `sendTo` function.
+  assert(socket.protocol != IPPROTO_TCP, "Cannot `sendTo` on a TCP socket")
+  assert(not socket.isClosed, "Cannot `sendTo` on a closed socket")
+
+  var sa: Sockaddr_storage
+  var sl: Socklen
+  toSockAddr(address, port, sa, sl)
+  let datasz = min(size, data.len())
```

Comment:
I think this should throw instead if the requested size is too large. Or maybe even better just call `setLen` on `data` instead.

Reply:
Actually, I should just remove the `size: int` argument, it's not on the other `sendTo` string variant. Likely I got it mixed in by copying from the raw pointer `sendTo`.